sdks/js/packages/core/react/components/organization/billing/cycle-switch/index.tsx

```diff
@@ -0,0 +1,185 @@
+import { Dialog, Flex, Text, Image, Separator, Button } from '@raystack/apsara';
+import Skeleton from 'react-loading-skeleton';
+import { useNavigate, useParams } from '@tanstack/react-router';
+import cross from '~/react/assets/cross.svg';
+import styles from '../../organization.module.css';
+import { useCallback, useEffect, useState } from 'react';
+import { useFrontier } from '~/react/contexts/FrontierContext';
+import { V1Beta1Plan } from '~/src';
+import { toast } from 'sonner';
+import { getPlanIntervalName, getPlanPrice } from '~/react/utils';
+import * as _ from 'lodash';
+import { usePlans } from '../../plans/hooks/usePlans';
+import dayjs from 'dayjs';
+import { DEFAULT_DATE_FORMAT } from '~/react/utils/constants';
+
+export function ConfirmCycleSwitch() {
+  const { activePlan, client, paymentMethod, config } = useFrontier();
+  const navigate = useNavigate({ from: '/billing/cycle-switch/$planId' });
+  const { planId } = useParams({ from: '/billing/cycle-switch/$planId' });
+  const dateFormat = config?.dateFormat || DEFAULT_DATE_FORMAT;
+
+  const [isPlanLoading, setIsPlanLoading] = useState(false);
+  const [nextPlan, setNextPlan] = useState<V1Beta1Plan>();
+  const [isCycleSwitching, setCycleSwitching] = useState(false);
+
+  const closeModal = useCallback(
+    () => navigate({ to: '/billing' }),
+    [navigate]
+  );
+
+  const {
+    checkoutPlan,
+    isLoading: isPlanActionLoading,
+    changePlan,
+    verifyPlanChange
+  } = usePlans();
+
+  const nextPlanPrice = nextPlan ? getPlanPrice(nextPlan) : { amount: 0 };
+  const isPaymentRequired =
+    _.isEmpty(paymentMethod) && nextPlanPrice.amount > 0;
+
+  const nextPlanIntervalName = getPlanIntervalName(nextPlan);
+
+  const nextPlanMetadata = nextPlan?.metadata as Record<string, number>;
+  const activePlanMetadata = activePlan?.metadata as Record<string, number>;
+
+  const isUpgrade =
```

Comment:
TODO for later: Let's move this to a common place?

```
func isUpgrade(plan1, plan2)
Return type: boolean
```

Our logic for determining whether a plan is an upgrade should be the same everywhere. A common method would help us enforce that.
Same for `isDowngrade`, `isPaymentMethodRequired` etc.

Reply:
yes, i tried to refactor it, but current PR got big due to that i removed the refactor changes.
right now the same logic is used at 2 places. i will refactor this in future

Reply:
Sounds good